### PR TITLE
Specify variables for which grad check is skipped in check_backward

### PR DESCRIPTION
**Pull request by wkentaro**

### CODE REVIEW

File: chainer/gradient_check.py [hunk 1]
```diff
@@ -184,6 +184,8 @@ def check_backward(func, x_data, y_grad, params=(),
             :func:`assert_allclose`.
         rtol (float): Relative tolerance to be passed to
             :func:`assert_allclose`.
+        no_grads (list of bool): Flag to skip variable for gradient assertion.
+            It should be same length as x_data.
```
Comment (unnonouno) [inline chainer/gradient_check.py]: Add backquotes beside `x_data`

File: chainer/gradient_check.py [hunk 2]
```diff
@@ -227,13 +229,19 @@ def f():
...
-            assert_allclose(gx, x.grad, atol=atol, rtol=rtol)
-            assert gx.dtype is x.grad.dtype
-        else:
+    if no_grads is None:
+        no_grads = [x.data.dtype.kind != 'f' for x in xs]
+    else:
+        if len(no_grads) != len(xs):
+            raise ValueError(
+                'Length of no_grads param and xs should be same.')
+    for skip, x in zip(no_grads, xs):
```
Comment (unnonouno) [inline chainer/gradient_check.py]: Use `six.moves.zip`

File: tests/chainer_tests/test_gradient_check.py [hunk 3]
```diff
@@ -339,5 +339,27 @@ def f(x, y):
...
+        x2 = numpy.array([0, 1], dtype='i')  # grad check for this is skipped
+        g1 = numpy.array([1], dtype='f')
+
+        def f(x, y):
+            s = Ident()(x)
+            return s,
+
+        gradient_check.check_backward(f, (x1, x2), g1)
+
+    def test_no_grads_option(self):
```
Comment (unnonouno) [inline tests/chainer_tests/test_gradient_check.py]: Does this test really check if the second argument is skipped?

Comment (wkentaro) [inline tests/chainer_tests/test_gradient_check.py]: I updated the commit to check if it raises error.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**